Update binary setup page - Closes # 1604 (#1641)
LiskHQ/lisk-docs: modules/ROOT/pages/setup/binary.adoc

= Binary setup
Christopher Braithwaite <christopher.braithwaite@lightcurve.io> Mona Bärenfänger <mona@lightcurve.io>
:description: How to install and setup a Lisk Core (binary distribution).
:toc: preamble
:toclevels: 5
:page-toclevels: 4

// External URLs
:url_set_path: https://opensource.com/article/17/6/set-path-linux
// Project URLs
:url_run_logging: lisk-docs::run-blockchain/logging.adoc
:url_zsch_set_path: https://linuxhint.com/add-a-directory-to-path-in-zsh/
:url_install_docker_mac: https://docs.docker.com/desktop/install/mac-install/

> The binary installation offers an easy and automated approach to set up Lisk Core.

Reviewer: ```suggestion
The binary installation offers an easy and automated approach to setting up Lisk Core.
```

> 

Reviewer: Just for the sake of sentence case, I would do the following:

Line 17: Pre-installation

Line 19: Supported platforms

Author: Good point...well spotted 👍

= Binary setup
Christopher Braithwaite <christopher.braithwaite@lightcurve.io> Mona Bärenfänger <mona@lightcurve.io>
:description: How to install and setup a Lisk Core (binary distribution).
:toc: preamble
:toclevels: 5
:page-toclevels: 4

// External URLs
:url_set_path: https://opensource.com/article/17/6/set-path-linux
// Project URLs
:url_run_logging: lisk-docs::run-blockchain/logging.adoc
:url_zsch_set_path: https://linuxhint.com/add-a-directory-to-path-in-zsh/
:url_install_docker_mac: https://docs.docker.com/desktop/install/mac-install/

The binary installation offers an easy and automated approach to setup Lisk Core.

== Pre-Installation

=== Supported Platforms
* Ubuntu 18.04 (LTS) x86_64
* Ubuntu 20.04 (LTS) x86_64
* MacOS x86_64
* MacOS ARM

// include::partial$create-a-new-user.adoc[]

[[install]]
== Installation

This section details how to install Lisk Core using pre-built application packages.
Once completed, you will have a functioning node on the Lisk Network.

> NOTE: When using a MAC with the Apple Silicon (M1 processor), please adhere to the relevant following setup instructions below:

Author: OK, I have removed the note 👍

=== Environment Setup requirements
. Clean the old Node.js installations (with arm64 binaries).
+
[source,bash]
----
rm -rf ~/.nvm
----
. Add a new session and set the iTerm to open in the **Intel** mode and then, set a shortcut key for opening iTerm in the aforementioned mode.
After that, open a terminal in Intel mode.

. Install *Homebrew* and *nvm*.
+
[source,bash]
----
/bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
wget -qO- https://raw.githubusercontent.com/nvm-sh/nvm/v0.39.0/install.sh | bash
----

. ARM Homebrew (for Redis).
+
[source,bash]
----
arch -arm64 /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"

> ----

Reviewer: Both of these points install Homebrew, so IMHO these two points should be converted into two tabs. Where one tab will be for **ARM** and the other one will be for Intel.

+
[source,bash]
----
if [[ $(arch) = arm64 ]]; 
then
    echo 'Detected Apple Silicon, ARM Homebrew has precedence'
    export PATH="/opt/homebrew/bin:/usr/local/bin:$PATH"
else
    echo 'Detected Intel CPU architecture, Intel Homebrew has precedence'
    export PATH="/usr/local/bin:/opt/homebrew/bin:$PATH"
----

==== Optional
. Check the mode in which the terminal is running.
+
[source,bash]
----
echo "Current architecture: $(arch)"
----

. Restart the terminal and ensure you are running it in the Intel mode.
+
[source,bash]
----
$ arch
i386
----

=== Install Rosetta, Homebrew, and NVM

. Install Rosetta (Intel x86 emulation layer).
+
[source,bash]
----
softwareupdate --install-rosetta"
----
. Install Homebrew in Apple Silicon (ARM) mode.
+
[source,bash]
----
/bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install.sh)"
----
. Install Homebrew in Intel mode.
+
[source,bash]
----
arch -x86_64 /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install.sh)"
----
. Install nvm.
+
[source,bash]
----
arch -x86_64 /bin/bash -c "wget -qO- https://raw.githubusercontent.com/nvm-sh/nvm/v0.39.0/install.sh | bash"

> ----

Reviewer: Here we install Homebrew and nvm again, I am not sure why there is repetition so please confirm which set of steps should be part of the guide. It is unclear at the moment which steps should be kept and which OS/platform these steps point towards.

It might also be possible to create two tabs for Mac M1 and Intel chip for installing Home brew
and another set of tabs for M1 and Intel Chips to install nvm.

Now it is possible to add these binaries to your path in the `.zsh_profile`

When using the latest MAC hardware this allows mixing the Intel software with the native ARM compilations.

[NOTE]
====
Please note, that some minimal housekeeping is required to ensure the right paths are included in the profile.

* Apple Silicon (ARM) brew writes binaries to the following: `/opt/homebrew/bin`

* Intel brew binaries are stored (as usual), in the following location: `/usr/local/Homebrew/bin`

* Preference should be given to Apple Silicon binaries over Intel where possible.
+
[source,bash]
----
export PATH=/opt/homebrew/bin:/usr/local/Homebrew/bin:$PATH
----

Remember to export the nvm path, (this should be performed automatically by the script).

====

=== Install Node.js
To install Node.js
[source,bash]
----
arch -x86_64 nvm install 16
----
TIP: For installing docker on M1 chips, check out the guide about {url_install_docker_mac}[installing Docker on a Mac^].

=== Install Python on an M1 Mac

> To install Python on an M1 Mac.

Reviewer: Do we only need Python for M1 Macs and not for Intel based chips?

Author: @nagdahimanshu Do you know if this is the case, and we only need Python for M1 Macs?

Author: @sameersubudhi I believe Himanshu is on holiday, so do you know if this is the case, and we only need Python for M1 Macs?

Reviewer 2: Yes, this instruction was specific to M1. In case of x86 Intel Macbook we didn't need this.

:url_zsch_set_path: https://linuxhint.com/add-a-directory-to-path-in-zsh/
:url_install_docker_mac: https://docs.docker.com/desktop/install/mac-install/

The binary installation offers an easy and automated approach to setting up Lisk Core.

== Pre-installation

=== Supported platforms
* Ubuntu 18.04 (LTS) x86_64
* Ubuntu 20.04 (LTS) x86_64
* MacOS x86_64
* MacOS ARM (Apple Silicon)

// include::partial$create-a-new-user.adoc[]

[[install]]
== Installation

This section details how to install Lisk Core using pre-built application packages.
Once completed, you will have a functioning node on the Lisk Network.

When using a MAC, please adhere to the following setup instructions:

=== Environment setup requirements
. Clean the old Node.js installations (with arm64 binaries).
+
[source,bash]
----
rm -rf ~/.nvm
----
. Add a new session and set the iTerm to open in the **Intel** mode. Now set a shortcut key (quick access), for opening the iTerm in the aforementioned mode.
Following that, open a terminal in Intel mode. Optionally, to verify that the terminal is now running in Intel mode, execute the following.
[source,bash]
----
$ arch
i386
----

> . Install *Homebrew*

Author: Yes, I have now fixed it 👍

//
// ----

ARM compatible binaries::
+
--

[source,bash]
----
arch -arm64 /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
----

--

Intel compatible binaries::
+
--
[source,bash]
----
arch -x86_64 /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install.sh)"
----
--
====
. Add Homebrew to your `~/.zshrc`
+
[source,bash]
----
if [[ $(arch) = arm64 ]];
then
    echo 'Detected Apple Silicon, ARM Homebrew has precedence'
    export PATH="/opt/homebrew/bin:/usr/local/bin:$PATH"
else
    echo 'Detected Intel CPU architecture, Intel Homebrew has precedence'
    export PATH="/usr/local/bin:/opt/homebrew/bin:$PATH"
----
. Install *nvm*
+

[tabs]
====

ARM::
+
--
[source,bash]
----
arch -arm64 /bin/bash -c "wget -qO- https://raw.githubusercontent.com/nvm-sh/nvm/v0.39.0/install.sh | bash"
----
--
Intel::
+
--
[source,bash]
----
arch -x86_64 /bin/bash -c "wget -qO- https://raw.githubusercontent.com/nvm-sh/nvm/v0.39.0/install.sh | bash"
----
--
====

=== Install Rosetta

. Install Rosetta (Intel x86 emulation layer).
+
[source,bash]
----
softwareupdate --install-rosetta

> ----

Author: OK, now done 👍

Christopher Braithwaite <christopher.braithwaite@lightcurve.io> Mona Bärenfänger <mona@lightcurve.io>
:description: How to install and setup a Lisk Core (binary distribution).
:toc: preamble
:toclevels: 5
:page-toclevels: 4

// External URLs
:url_set_path: https://opensource.com/article/17/6/set-path-linux
// Project URLs
:url_run_logging: lisk-docs::run-blockchain/logging.adoc
:url_zsch_set_path: https://linuxhint.com/add-a-directory-to-path-in-zsh/
:url_install_docker_mac: https://docs.docker.com/desktop/install/mac-install/

The binary installation offers an easy and automated approach to setting up Lisk Core.

== Pre-installation

=== Supported platforms
* Ubuntu 18.04 (LTS) x86_64
* Ubuntu 20.04 (LTS) x86_64
* MacOS x86_64
* MacOS ARM

// include::partial$create-a-new-user.adoc[]

[[install]]
== Installation

This section details how to install Lisk Core using pre-built application packages.
Once completed, you will have a functioning node on the Lisk Network.

When using a MAC, please adhere to the following setup instructions:

=== Environment setup requirements
. Clean the old Node.js installations (with arm64 binaries).
+
[source,bash]
----
rm -rf ~/.nvm
----

> . Add a new session and set the iTerm to open in the **Intel** mode. Now set a shortcut key for opening the iTerm in the aforementioned mode.

Reviewer 2: ```suggestion
. Add a new session and set the iTerm to open in the **Intel** mode. Now set a shortcut key (quick access) for opening the iTerm in the aforementioned mode.
```

> Following that, open a terminal in Intel mode.

Reviewer 2: ```suggestion
Following that, open a terminal in Intel mode. Optionally, to verify that the terminal is now running in Intel mode, execute the following.
[source,bash]
----
$ arch
i386
----
```

. Install *Homebrew*
+
[tabs]
====

// Ubuntu::
// +
// [source,bash]
// ----
// /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
//
// ----

> ARM::
> +
> --
> For Redis and Apple Silicon use the following command:

Reviewer 2: ```suggestion
ARM compatible binaries::
+
--
```

[source,bash]
----
arch -arm64 /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
----

--

> Intel::
> +
> --
> For Intel, use the following command:

Reviewer 2: ```suggestion
Intel compatible binaries::
+
--
```

[source,bash]
----
arch -x86_64 /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install.sh)"
----
--
====
. Add Homebrew to your `~/.zshrc`
+
[source,bash]
----
if [[ $(arch) = arm64 ]];
then
    echo 'Detected Apple Silicon, ARM Homebrew has precedence'
    export PATH="/opt/homebrew/bin:/usr/local/bin:$PATH"
else
    echo 'Detected Intel CPU architecture, Intel Homebrew has precedence'
    export PATH="/usr/local/bin:/opt/homebrew/bin:$PATH"
----
. Install *nvm*
+

[tabs]
====

ARM::
+
--
[source,bash]
----

> wget -qO- https://raw.githubusercontent.com/nvm-sh/nvm/v0.39.0/install.sh | bash

Reviewer 2: ```suggestion
arch -arm64 /bin/bash -c "wget -qO- https://raw.githubusercontent.com/nvm-sh/nvm/v0.39.0/install.sh | bash"
```

----
--
Intel::
+
--
[source,bash]
----
arch -x86_64 /bin/bash -c "wget -qO- https://raw.githubusercontent.com/nvm-sh/nvm/v0.39.0/install.sh | bash"
----
--
====

> 
> 
> ==== Optional
> . Check the mode in which the terminal is running.
> +
> [source,bash]
> ----
> echo "Current architecture: $(arch)"
> ----
> 
> . Restart the terminal and ensure you are running it in the Intel mode.
> +
> [source,bash]
> ----
> $ arch
> i386
> ----

Reviewer 2: This entire section seems irrelevant can be removed. However, the verification step can be added to the second point after we ask the user to open iTerm in Intel mode.
```suggestion
```

> === Install Rosetta

Reviewer 2: The section about installing rosetta must be the first step.

Author: Now done 👍

. Install Rosetta (Intel x86 emulation layer).
+
[source,bash]
----

> softwareupdate --install-rosetta"

Reviewer 2: ```suggestion
softwareupdate --install-rosetta
```

----
// . Install Homebrew in Apple Silicon (ARM) mode.
// +
// [source,bash]
// ----
// /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install.sh)"
//----

==== Add to $PATH

Now it is possible to add these binaries to your path in the `.zsh_profile`

When using the latest MAC hardware this allows mixing the Intel software with the native ARM compilations.

[NOTE]
====
Please note, that some minimal housekeeping is required to ensure the right paths are included in the profile.

* Apple Silicon (ARM) brew writes binaries to the following: `/opt/homebrew/bin`

* Intel brew binaries are stored (as usual), in the following location: `/usr/local/Homebrew/bin`

* Preference should be given to Apple Silicon binaries over Intel where possible.

Remember to export the nvm path, (this should be performed automatically by the script).

====

=== Install Node.js
To install Node.js
[source,bash]
----
arch -x86_64 nvm install 16
----
TIP: For installing docker on M1 chips, check out the guide about {url_install_docker_mac}[installing Docker on a Mac^].

> === Install Python (M1 Macs only)
> To install Python on an M1 Mac, execute the following commands:

Reviewer 2: ```suggestion
=== Install Python (Macs running on Apple Silicon)
To install Python on a Mac powered by Apple Silicon, execute the following commands:
```

Once completed, you will have a functioning node on the Lisk Network.

NOTE: When using a MAC with Apple Silicon (M1 processor) please adhere to the relevant following setup instructions below:

=== Environment Setup requirements
Clean the old Node.js installations (with arm64 binaries)
[source,bash]
----
rm -rf ~/.nvm
----
Add a new session and set the iTerm to open the Intel mode and set a shortcut key
Open a terminal in the Intel mode

==== Install Homebrew and nvm
[source,bash]
----
/bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
wget -qO- https://raw.githubusercontent.com/nvm-sh/nvm/v0.39.0/install.sh | bash
----
==== ARM Homebrew (for Redis)
[source,bash]
----
arch -arm64 /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
----
Add the following lines to your ~/.zshrc
[source,bash]
----
if [[ $(arch) = arm64 ]]; then
#   echo 'Detected Apple Silicon, ARM Homebrew has precedence'
   export PATH="/opt/homebrew/bin:/usr/local/bin:$PATH"
else
#   echo 'Detected Intel CPU architecture, Intel Homebrew has precedence'
   export PATH="/usr/local/bin:/opt/homebrew/bin:$PATH"

> ----

Reviewer: The formatting of this section should be improved and should match [this section](https://liskhq.github.io/lisk-docs/beta/build-blockchain/plugin/plugin-endpoints.html#testing-helloinfoplugin).
- A numbered list with bold text at the start will improve the text a lot.

==== Optional

> [source,bash]

Reviewer: ```suggestion
Check the mode in which the terminal is running.

[source,bash]
```

----
echo "Current architecture: $(arch)"
----
Restart the terminal and ensure you are running it in the Intel mode.
[source,bash]
----
$ arch
i386
----

> Install - Node.js
> [source,bash]
> ----
> arch -x86_64 nvm install 12
> ----

Reviewer: Since we already have a section to install node.js, we can remove it from here.
```suggestion

```

> Install docker for m1 by using google search for “docker install m1” and then run it.

Reviewer: This should be a **TIP**. Please add the appropriate variable for the URL pasted below and add the hyperlink to the guide in the TIP.

Also, please move the updated code to replace content on **Line 127**
```suggestion
TIP: For installing docker on m1 chips, check out the guide about [Installing on Mac](https://docs.docker.com/desktop/install/mac-install/)
```

==== Install Rosetta, Homebrew, and NVM

Install Rosetta (Intel x86 emulation layer).

[source,bash]
----
softwareupdate --install-rosetta"
----
Install Homebrew in Apple Silicon (ARM) mode
[source,bash]
----
/bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install.sh)"
----
Install Homebrew in Intel mode
[source,bash]
----
arch -x86_64 /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install.sh)"
----
Install nvm
----
arch -x86_64 /bin/bash -c "wget -qO- https://raw.githubusercontent.com/nvm-sh/nvm/v0.39.0/install.sh | bash"

> ----

Reviewer: The formatting of this section should be improved and should match [this section](https://liskhq.github.io/lisk-docs/beta/build-blockchain/plugin/plugin-endpoints.html#testing-helloinfoplugin).
- A numbered list with guiding text, followed by the commands.

Author: Done 👍

Now it is possible to add these binaries to your path in the .zsh_profile

When using the latest MAC hardware this allows mixing the Intel software with the native ARM compilations.

[TIP]
Please note, that some minimal housekeeping is required to ensure the right paths are included in the profile.

Apple Silicon (ARM) brew writes binaries to the following: /opt/homebrew/bin

Intel brew binaries are stored (as usual), in the following location: /usr/local/Homebrew/bin

Preference should be given to Apple Silicon binaries over Intel where possible.

[source,bash]
----
export PATH=/opt/homebrew/bin:/usr/local/Homebrew/bin:$PATH
----

> Remember to export the nvm path, (this should be performed automatically by the script).

Reviewer: This part should be converted into a `NOTE`

==== Install Node.js
To install Node.js
[source,bash]
----

> arch -x86_64 nvm install 16

Reviewer: Should be the latest version of the nodeJS, which is `18`. Please confirm the version

Author: @nagdahimanshu Can you please confirm the Node.js version is 18?

Reviewer 3: We are currently using Nodejs `v16.15.0` version.

// # The new Macs allow you to mix Intel software with native ARM compilations.

// # Some minimal housekeeping is required to ensure the right paths are included in the profile.

// # Apple Silicon (ARM) brew writes binaries to /opt/homebrew/bin

// # Intel brew binaries are stored (as usual) in /usr/local/Homebrew/bin

// Preference should be given to Apple Silicon binaries over Intel where possible

// export PATH=/opt/homebrew/bin:/usr/local/Homebrew/bin:$PATH

// # Remember to export the nvm path (should be done automatically by the script)
// ----

// ====
// .Install Node.js
// [%collapsible]
// ====
// [source,js]
// ----

// Install - Node.js
// arch -x86_64 nvm install 16

// Install docker for m1 by google search “docker install m1” and run it

// ----
// ====
// .Install Python on M1 Mac
// [%collapsible]
// ====
// [source,js]
// ----
// brew install pyenv
// pyenv install 2.7.18
// export PATH="$(pyenv root)/shims:${PATH}"
// echo 'PATH=$(pyenv root)/shims:$PATH' >> ~/.zshrc
// pyenv global 2.7.18
// ----

// ====
// Is python v 2.7.18 still correct as the latest version is 3.11.1?
==== Download and execute the installation script

> Download Lisk Core:

Reviewer: ```suggestion
Download Lisk Core
```

* Ubuntu 18.04 (LTS) x86_64
* Ubuntu 20.04 (LTS) x86_64
* MacOS x86_64

> * MacOS ARM

Reviewer: Is this the updated list of supported platforms?

Author: I spoke with Himanshu and he has now confirmed this is correct 👍

- Install nvm
arch -x86_64 /bin/bash -c "wget -qO- https://raw.githubusercontent.com/nvm-sh/nvm/v0.39.0/install.sh | bash"

# Now you can add these binaries to your path in .zsh_profile

# The new Macs allow you to mix Intel software with native ARM compilations.

# Some minimal housekeeping is required to ensure the right paths are included in the profile.

# Apple Silicon (ARM) brew writes binaries to /opt/homebrew/bin

# Intel brew binaries are stored (as usual) in /usr/local/Homebrew/bin

Preference should be given to Apple Silicon binaries over Intel where possible

export PATH=/opt/homebrew/bin:/usr/local/Homebrew/bin:$PATH

# Remember to export the nvm path (should be done automatically by the script)
----

====
.Install Node.js
[%collapsible]
====
[source,js]
----

Install - Node.js
arch -x86_64 nvm install 16

Install docker for m1 by google search “docker install m1” and run it

----
====
.Install Python on M1 Mac
[%collapsible]
====
[source,js]
----
brew install pyenv
pyenv install 2.7.18
export PATH="$(pyenv root)/shims:${PATH}"
echo 'PATH=$(pyenv root)/shims:$PATH' >> ~/.zshrc

> pyenv global 2.7.18

Reviewer: Please separate the installation steps from the commands that should be run to install dependencies.

----
====
// Is python v 2.7.18 still correct as the latest version is 3.11.1?
=== Download and execute the installation script

> Download Lisk Core:

Reviewer: **Line 139**
For uniformity, the `tabs` delimiter should be `=====` instead of `====`

https://github.com/asciidoctor/asciidoctor-tabs#syntax

> Download Lisk Core:

Reviewer: **Line 185 - 206**

Since we also support macOS and it now contains `zshrc` terminal, please add similar steps for the `zshrc` terminal as well. You can see the following URL for guidance.
https://liskhq.github.io/lisk-docs/beta/build-blockchain/create-blockchain-app.html#using-the-application-cli-globally

* Ubuntu 18.04 (LTS) x86_64
* Ubuntu 20.04 (LTS) x86_64
* MacOS x86_64
* MacOS ARM

// include::partial$create-a-new-user.adoc[]

[[install]]
== Installation

This section details how to install Lisk Core using pre-built application packages.
Once completed, you will have a functioning node on the Lisk Network.

When using a MAC, please adhere to the following setup instructions:

> === Environment Setup requirements

Reviewer: ```suggestion
=== Environment setup requirements
```

----
--
====

==== Optional
. Check the mode in which the terminal is running.
+
[source,bash]
----
echo "Current architecture: $(arch)"
----

. Restart the terminal and ensure you are running it in the Intel mode.
+
[source,bash]
----
$ arch
i386
----

=== Install Rosetta

. Install Rosetta (Intel x86 emulation layer).
+
[source,bash]
----
softwareupdate --install-rosetta"
----
// . Install Homebrew in Apple Silicon (ARM) mode.
// +
// [source,bash]
// ----
// /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install.sh)"
//----

==== Add to $PATH

Now it is possible to add these binaries to your path in the `.zsh_profile`

When using the latest MAC hardware this allows mixing the Intel software with the native ARM compilations.

[NOTE]
====
Please note, that some minimal housekeeping is required to ensure the right paths are included in the profile.

* Apple Silicon (ARM) brew writes binaries to the following: `/opt/homebrew/bin`

* Intel brew binaries are stored (as usual), in the following location: `/usr/local/Homebrew/bin`

* Preference should be given to Apple Silicon binaries over Intel where possible.

> +
> [source,bash]
> ----
> ----

Reviewer: The code block gets rendered, so it should be removed.
```suggestion

```

* Ubuntu 18.04 (LTS) x86_64
* Ubuntu 20.04 (LTS) x86_64
* MacOS x86_64
* MacOS ARM

// include::partial$create-a-new-user.adoc[]

[[install]]
== Installation

This section details how to install Lisk Core using pre-built application packages.
Once completed, you will have a functioning node on the Lisk Network.

> When using a MAC with the Apple Silicon (M1 processor), please adhere to the relevant following setup instructions below:

Reviewer: ```suggestion
When using a MAC, please adhere to the following setup instructions:
```

=== Environment Setup requirements
. Clean the old Node.js installations (with arm64 binaries).
+
[source,bash]
----
rm -rf ~/.nvm
----
. Add a new session and set the iTerm to open in the **Intel** mode and then, set a shortcut key for opening iTerm in the aforementioned mode.
After that, open a terminal in Intel mode.

. Install *Homebrew*
+
[tabs]
====

// Ubuntu::
// +
// [source,bash]
// ----
// /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
//
// ----

ARM::
+
--
For Redis and Apple Silicon use the following command:

[source,bash]
----
arch -arm64 /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
----

> For Apple Silicon add the following lines to your `~/.zshrc`
> 
> [source,bash]
> ----
> if [[ $(arch) = arm64 ]];
> then
>     echo 'Detected Apple Silicon, ARM Homebrew has precedence'
>     export PATH="/opt/homebrew/bin:/usr/local/bin:$PATH"
> else
>     echo 'Detected Intel CPU architecture, Intel Homebrew has precedence'
>     export PATH="/usr/local/bin:/opt/homebrew/bin:$PATH"
> ----

Reviewer: I think this can be moved to a separate point, as suggested below
```suggestion

```

--

Intel::
+

> --

Reviewer: ```suggestion
--
For Intel, use the following command:
```

[source,bash]
----
arch -x86_64 /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install.sh)"
----
--
====

> 

Reviewer: ```suggestion
. Add Homebrew to your `~/.zshrc`
+
[source,bash]
----
if [[ $(arch) = arm64 ]];
then
    echo 'Detected Apple Silicon, ARM Homebrew has precedence'
    export PATH="/opt/homebrew/bin:/usr/local/bin:$PATH"
else
    echo 'Detected Intel CPU architecture, Intel Homebrew has precedence'
    export PATH="/usr/local/bin:/opt/homebrew/bin:$PATH"
----
```

----
--
====

==== Optional
. Check the mode in which the terminal is running.
+
[source,bash]
----
echo "Current architecture: $(arch)"
----

. Restart the terminal and ensure you are running it in the Intel mode.
+
[source,bash]
----
$ arch
i386
----

=== Install Rosetta

. Install Rosetta (Intel x86 emulation layer).
+
[source,bash]
----
softwareupdate --install-rosetta"
----
// . Install Homebrew in Apple Silicon (ARM) mode.
// +
// [source,bash]
// ----
// /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install.sh)"
//----

==== Add to $PATH

Now it is possible to add these binaries to your path in the `.zsh_profile`

When using the latest MAC hardware this allows mixing the Intel software with the native ARM compilations.

[NOTE]
====
Please note, that some minimal housekeeping is required to ensure the right paths are included in the profile.

* Apple Silicon (ARM) brew writes binaries to the following: `/opt/homebrew/bin`

* Intel brew binaries are stored (as usual), in the following location: `/usr/local/Homebrew/bin`

* Preference should be given to Apple Silicon binaries over Intel where possible.
+
[source,bash]
----

> export PATH=/opt/homebrew/bin:/usr/local/Homebrew/bin:$PATH

Reviewer: Should be removed as we already tell the reader to add homebrew to the `$PATH` variable.
```suggestion

```

Author: Good point 👍

----

Remember to export the nvm path, (this should be performed automatically by the script).

====

=== Install Node.js
To install Node.js
[source,bash]
----
arch -x86_64 nvm install 16
----
TIP: For installing docker on M1 chips, check out the guide about {url_install_docker_mac}[installing Docker on a Mac^].

> === Install Python on an M1 Mac
> To install Python on an M1 Mac.

Reviewer: ```suggestion
=== Install Python (M1 Macs only)
To install Python on an M1 Mac, execute the following commands:
```

This section details how to install Lisk Core using pre-built application packages.
Once completed, you will have a functioning node on the Lisk Network.
--
NOTE: When using a MAC with Apple Silicon (M1) please adhere to the relevant following setup instructions below:

> .Environment Setup requirements

Author: Many thanks Sameer! I will await your feedback 👍

[source,bash]
----
curl -o lisk-core.tar.gz https://downloads.lisk.com/lisk/testnet/4.0.0/lisk-core-v4.0.0-linux-x64.tar.gz
----

*MacOS:*

[source,bash]
----
curl -o lisk-core.tar.gz https://downloads.lisk.com/lisk/testnet/4.0.0/lisk-core-v4.0.0-darwin-x64.tar.gz
----
--
====

> Unpack it and add it to the system path, in order to use it in the terminal:

Reviewer 2: ```suggestion
Unpack and add it to the system path, in order to use it in the terminal:
```
